packages/cloud-cognitive/src/components/CreateTearsheetNarrow/CreateTearsheetNarrow.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright IBM Corp. 2021, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// Import portions of React that are needed.
+import React from 'react';
+
+// Other standard imports.
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { Form } from 'carbon-components-react';
+import { pkg } from '../../settings';
+import { TearsheetNarrow } from '../Tearsheet/TearsheetNarrow';
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--create-tearsheet-narrow`;
+const componentName = 'CreateTearsheetNarrow';
+
+// NOTE: the component SCSS is not imported here: it is rolled up separately.
+
+/**
+ * Use a narrow tearsheet as an alternative to a modal when there is scrolling.
+ * Use when the form fields can be broken down into sections using section headers.
+ */
+export let CreateTearsheetNarrow = React.forwardRef(
+  (
+    {
+      children,
+      className,
+      description,
+      disableSubmit,
+      formDescription,
+      formTitle,
+      label,
+      open,
+      onRequestClose,
+      onRequestSubmit,
+      primaryButtonText,
+      secondaryButtonText,
+      selectorPrimaryFocus,
+      title,
+      verticalPosition,
+      ...rest
+    },
+    ref
+  ) => {
+    const actions = [
+      {
+        label: primaryButtonText,
+        onClick: onRequestSubmit,
+        kind: 'primary',
+        disabled: disableSubmit,
+      },
+      {
+        label: secondaryButtonText,
+        onClick: onRequestClose,
+        kind: 'secondary',
+      },
+    ];
+    return (
+      <TearsheetNarrow
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        title={title}
+        description={description}
+        className={cx(blockClass, className)}
+        actions={actions}
+        open={open}
+        ref={ref}
+        onClose={onRequestClose}
+        label={label}
+        selectorPrimaryFocus={selectorPrimaryFocus}
+        verticalPosition={verticalPosition}
+        role="presentation">
+        <h3
+          className={`${blockClass}__form-title-text ${blockClass}__content-text`}>
```

Comment:
personally, when i have multiple classes that are like `${blockClass}__element-text ${blockClass}__element-container ${blockClass}__element-header` etc i usually just add an additional class variable so it ends up just being `const elementClass = ${blockClass}__element; ... ${elementClass}-text ${elementClass}-container ${elementClass}-element`. not a big deal. maybe just a personal preference thing, but just putting that out there.

Reply:
Yeah, I like that better too. I'll make a separate PR to add this in so we can get an initial version of this component out there